manager/dispatcher/assignments.go

```diff
@@ -245,3 +259,21 @@ func (a *assignmentSet) message() api.AssignmentsMessage {
 
 	return message
 }
+
+// populateSecretFromPlugin populates the secret value for the given specification using the secret plugin subsystem.
+func (a *assignmentSet) populateSecretFromDriver(spec *api.SecretSpec) error {
+	if spec == nil || spec.Driver.Name == "" {
```

Comment:
Thanks, I've added shared code to validate the secret payload, after the value is populated.